test/gateway/redeem_proxy/self_destruct.js

```diff
@@ -0,0 +1,36 @@
+const RedeemerProxy = artifacts.require('RedeemerProxy');
+
+const web3 = require('../../test_lib/web3');
+const Utils = require('../../test_lib/utils');
+
+contract('RedeemerProxy.selfDestruct() ', (accounts) => {
+  let composer;
+  let owner;
+  let proxy;
+
+  beforeEach(async () => {
+    composer = accounts[1];
+    owner = accounts[2];
+    proxy = await RedeemerProxy.new(owner, { from: composer });
+  });
+
+  it('should successfully self destruct', async () => {
+    await proxy.selfDestruct({ from: composer });
+
+    const code = await web3.eth.getCode(proxy.address);
```

Comment:
Please check that before calling `selfDestruct` the `proxy.address` had non-null code.